orttraining/orttraining/python/ort_trainer.py

```diff
@@ -773,7 +775,15 @@ def state_dict(self):
             if n.name not in torch_state:
                 torch_state[n.name] = torch.from_numpy(numpy_helper.to_array(n))
 
-        return torch_state
+        # Need to remove redundant initializers and name suffices to map back to original torch state names
+        torch_state_to_return = {}
```

Comment:
I remember python have a syntax like this 

torch_state_to_return = [x if x in self.original_model_state_keys]

Reply:
sure, i can change it to a dictionary comprehension.